pageserver/src/tenant/secondary/downloader.rs

```diff
@@ -498,6 +551,24 @@ impl<'a> TenantDownloader<'a> {
 
         tracing::debug!("Wrote local heatmap to {}", heatmap_path);
 
+        // Clean up any local layers that aren't in the heatmap.  We do this first for all timelines, on the general
+        // principle that deletions should be done before writes whereever possible, and so that we can use this
+        // phase to initialize our SecondaryProgress.
+        {
+            let heatmap_stats = heatmap.get_stats();
+            // We will construct a progress object, and then populate its initial "downloaded" numbers
+            // while iterating through local layer state in [`Self::prepare_timelines`]
+            let mut progress_init = SecondaryProgress {
+                layers_total: heatmap_stats.layers,
+                bytes_total: heatmap_stats.bytes,
+                heatmap_mtime: Some(heatmap_mtime),
+                layers_downloaded: 0,
+                bytes_downloaded: 0,
+            };
+            self.prepare_timelines(&heatmap, &mut progress_init).await?;
+            *self.secondary_state.progress.lock().unwrap() = progress_init;
```

Comment:
Hmm.  My starting instinct here was to keep them separate, because up to this point `detail` is only ever accessed from `downloader.rs` -- it's a "never contended" lock.  Whereas `progress` can in principal be contended if we let the HTTP API query it at will.  However, it's a sync lock so the contention doesn't matter hugely, the lock will only ever be held for microseconds.  The concurrency side of this is largely a non-issue: secondaries are intentionally quite slow/rare/lightweight things.

I don't feel terribly strongly: but I'm inclined to leave it as two members just so that we don't end up with a Mutex<tuple>, or having to invent another struct for ProgressAndDetail (just adding it to detail would mean broadening the visibility of that struct outside downloader.rs, which I would like to avoid.)